add_libmagic.sh

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env bash
+
+set -euxo pipefail
+
+install_source() {
+    # install from source
+    # https://www.darwinsys.com/file/
+    # https://github.com/file/file/blob/FILE5_45/INSTALL#L51
+    (
+        version="file-5.45" &&
+        tmpfile="$(mktemp)" &&
+        curl -sSLo "${tmpfile}" "https://astron.com/pub/file/${version}.tar.gz" &&
+        tar xvf "${tmpfile}" &&
+        cd "${version}" &&
+        ./configure &&
+        make &&
```

Comment:
Developer's dependencies to use "make" and similar commands are not shipped by default in some distros, especially in those that are usually used by Docker. I'd suggest to ensure that necessary dependencies are installed first.

Reply:
I wished `set -euxo pipefail` also effects commands inside function definitions. then we could remove `&&` from every line here (and the subshell parentheses here and the `||` on L21), and we could test `which make` on top of this function or similar.

now, it simply falls back to `install_precompiled` (L69) when curl or make or compilation fails (error will be in output).

if make is not available, in the current setup we will only have done an unnecessary curl. not the worst but could be avoided.

any ideas?

Reply:
> I wished `set -euxo pipefail` also effects commands inside function definitions. then we could remove `&&` from every line here (and the subshell parentheses here and the `||` on L21), and we could test `which make` on top of this function or similar.
> 
> any ideas?

Just use it [inside](https://unix.stackexchange.com/questions/405382/set-e-inside-a-bash-function) of a function.

You made me think about other topic when you mentioned that `curl` can be executed. There is no cleanup on any step, it might be logical to make a ZIP file cleanup from external function and do it regardless of the outcome. Same for windows. Pseudocode:
install_stuff(){
 real_install()
 rm potential_zip
}